### PR TITLE
Code style black

### PR DESCRIPTION
**Pull request by wkerzendorf**
Switching code-style to black - one `subpackage` at a time.

### CODE REVIEW

File: tardis/io/atom_data/util.py [hunk 1]
```diff
@@ -26,17 +30,23 @@ def resolve_atom_data_fname(fname):
...
-
+        raise IOError(
+            "Atom Data {0} not found in path or in TARDIS data repo - it is available as download:\n"
+            "from tardis.io.atom_data import download_atom_data\n"
+            "download_atom_data('{1}')".format(fname, atom_data_name)
+        )
+
+    raise IOError(
+        "Atom Data {0} is not found in current path or in TARDIS data repo. {1} is also not a standard known"
+        "TARDIS atom dataset.".format(fname, atom_data_name)
```
Comment (jaladh-singhal) [inline tardis/io/atom_data/util.py:51]:
```suggestion
        f"Atom Data {fname} is not found in current path or in TARDIS data repo. {atom_data_name} is also 
        "not a standard known TARDIS atom dataset."
```

File: tardis/io/config_reader.py [hunk 2]
```diff
@@ -79,10 +82,9 @@ def from_yaml(cls, fname):
         try:
             yaml_dict = yaml_load_file(fname)
         except IOError as e:
-            logger.critical('No config file named: %s', fname)
+            logger.critical("No config file named: %s", fname)
```
Comment (wkerzendorf) [inline tardis/io/config_reader.py:85]:
```suggestion
            logger.critical(f"No config file named: {fname}")
```

File: tardis/io/atom_data/util.py [hunk 3]
```diff
@@ -26,17 +30,23 @@ def resolve_atom_data_fname(fname):
...
-
+        raise IOError(
+            "Atom Data {0} not found in path or in TARDIS data repo - it is available as download:\n"
+            "from tardis.io.atom_data import download_atom_data\n"
+            "download_atom_data('{1}')".format(fname, atom_data_name)
+        )
+
+    raise IOError(
+        f"Atom Data {fname} is not found in current path or in TARDIS data repo. {atom_data_name} is also 
+        "not a standard known TARDIS atom dataset."
```
Comment (jaladh-singhal) [inline tardis/io/atom_data/util.py:51]:
```suggestion
        f"Atom Data {fname} is not found in current path or in TARDIS data repo. {atom_data_name} "
        "is also not a standard known TARDIS atom dataset."
```

File: tardis/io/config_reader.py [hunk 4]
```diff
@@ -230,24 +241,25 @@ class Configuration(ConfigurationNameSpace):
...
     def from_yaml(cls, fname, *args, **kwargs):
         try:
-            yaml_dict = yaml_load_file(fname,
-                                       loader=kwargs.pop('loader', YAMLLoader))
+            yaml_dict = yaml_load_file(
+                fname, loader=kwargs.pop("loader", YAMLLoader)
+            )
         except IOError as e:
-            logger.critical('No config file named: %s', fname)
+            logger.critical("No config file named: %s", fname)
```
Comment (wkerzendorf) [inline tardis/io/config_reader.py:248]:
```suggestion
            logger.critical(f"No config file named: {fname}")
```

File: tardis/io/decay.py [hunk 5]
```diff
@@ -22,38 +23,38 @@ def _constructor(self):
     def _update_material(self):
         self.comp_dicts = [dict() for i in range(len(self.columns))]
         for (atomic_number, mass_number), abundances in self.iterrows():
-            nuclear_symbol = '%s%d'.format(nucname.name(atomic_number),
-                                           mass_number)
+            nuclear_symbol = "%s%d".format(
+                nucname.name(atomic_number), mass_number
+            )
```
Comment (wkerzendorf) [inline tardis/io/decay.py:28]:
```suggestion
            nuclear_symbol = f"{nucname.name(atomic_number)}{mass_number}"
            )
```

File: tardis/io/util.py [hunk 6]
```diff
@@ -376,10 +381,14 @@ def download_from_url(url, dst):
...
-        unit='B', unit_scale=True, desc=url.split('/')[-1])
+        total=file_size,
+        initial=first_byte,
+        unit="B",
+        unit_scale=True,
+        desc=url.split("/")[-1],
+    )
     req = requests.get(url, headers=header, stream=True)
-    with(open(dst, 'ab')) as f:
+    with (open(dst, "ab")) as f:
```
Comment (jaladh-singhal) [inline tardis/io/util.py:391]:
```suggestion
    with open(dst, "ab") as f:
```

File: tardis/plasma/properties/atomic.py [hunk 7]
```diff
@@ -142,44 +175,44 @@ class AtomicMass(ProcessingPlasmaProperty):
...
             return ionization_data
         else:
             raise IncompleteAtomicData(
-                    'ionization data for the ion ({}, {})'.format(
-                            str(counts.index[counts.index != counts]),
-                            str(counts[counts.index != counts])
-                            )
-                    )
+                "ionization data for the ion ({}, {})".format(
+                    str(counts.index[counts.index != counts]),
```
Comment (wkerzendorf) [inline tardis/plasma/properties/atomic.py:207]: need to change that.

File: tardis/plasma/properties/general.py [hunk 8]
```diff
@@ -30,20 +40,26 @@ def __init__(self, plasma_parent):
...
-    latex_name = ('g_{\\textrm{electron}}',)
-    latex_formula = ('\\Big(\\dfrac{2\\pi m_{e}/\
-                     \\beta_{\\textrm{rad}}}{h^2}\\Big)^{3/2}',)
+
+    outputs = ("g_electron",)
+    latex_name = ("g_{\\textrm{electron}}",)
+    latex_formula = (
+        "\\Big(\\dfrac{2\\pi m_{e}/\
+                     \\beta_{\\textrm{rad}}}{h^2}\\Big)^{3/2}",
+    )
```
Comment (wkerzendorf) [inline tardis/plasma/properties/general.py:56]:
```suggestion
    latex_formula = (r"\Big(\dfrac{2\pi m_{e}/\beta_{\textrm{rad}}}{h^2}\Big)^{3/2}",)
```

File: tardis/plasma/properties/general.py [hunk 9]
```diff
@@ -52,10 +65,13 @@ class ThermalGElectron(GElectron):
...
-    outputs = ('thermal_g_electron',)
-    latex_name = ('g_{\\textrm{electron_thermal}}',)
-    latex_formula = ('\\Big(\\dfrac{2\\pi m_{e}/\
-                     \\beta_{\\textrm{electron}}}{h^2}\\Big)^{3/2}',)
+
+    outputs = ("thermal_g_electron",)
+    latex_name = ("g_{\\textrm{electron_thermal}}",)
+    latex_formula = (
+        "\\Big(\\dfrac{2\\pi m_{e}/\
+                     \\beta_{\\textrm{electron}}}{h^2}\\Big)^{3/2}",
```
Comment (jaladh-singhal) [inline tardis/plasma/properties/general.py:73]:
```suggestion
        r"\Big(\dfrac{2\pi m_{e}/\beta_{\textrm{electron}}}{h^2}\Big)^{3/2}",
```

File: tardis/plasma/properties/nlte.py [hunk 10]
```diff
@@ -4,128 +4,194 @@
...
+            t_electrons,
+            delta,
+            g_electron,
+            beta_rad,
+            ionization_data,
+            g,
+        )
+        #        unnormalised = helium_population.sum()
+        #        normalised = helium_population.mul(number_density.ix[2] /
+        # unnormalised)
```
Comment (jaladh-singhal) [inline tardis/plasma/properties/nlte.py:105]:
```suggestion
        #        normalised = helium_population.mul(number_density.ix[2] / unnormalised)
```

File: tardis/plasma/properties/partition_function.py [hunk 11]
```diff
@@ -116,25 +132,29 @@ class LevelBoltzmannFactorNLTE(ProcessingPlasmaProperty):
...
+        ):
             return LevelBoltzmannFactorNLTECoronal
         elif nlte_conf.coronal_approximation and nlte_conf.classical_nebular:
-            raise PlasmaConfigError('Both coronal approximation and '
-                                    'classical nebular specified in the '
-                                    'config.')
+            raise PlasmaConfigError(
+                "Both coronal approximation and "
+                "classical nebular specified in the "
+                "config."
```
Comment (jaladh-singhal) [inline tardis/plasma/properties/partition_function.py:156]:
```suggestion
                "Both coronal approximation and classical nebular specified in the config."
```

File: tardis/plasma/properties/partition_function.py [hunk 12]
```diff
@@ -148,19 +168,26 @@ def __init__(self, plasma_parent):
...
+        previous_electron_densities,
+        g,
+    ):
         """
         The core of the NLTE calculation, used with all possible config.
         options.
         """
         for species in nlte_data.nlte_species:
-            logger.info('Calculating rates for species %s', species)
+            logger.info("Calculating rates for species %s", species)
```
Comment (wkerzendorf) [inline tardis/plasma/properties/partition_function.py:186]:
```suggestion
            logger.info(f"Calculating rates for species {species}")
```

File: tardis/plasma/standard_plasmas.py [hunk 13]
```diff
@@ -49,133 +60,156 @@ def assemble_plasma(config, model, atom_data=None):
...
+            j_blue_estimator=None,
+        )
+        property_kwargs[JBluesDetailed] = {"w_epsilon": config.plasma.w_epsilon}
     else:
-        raise ValueError('radiative_rates_type type unknown - %s',
-                         config.plasma.radiative_rates_type)
+        raise ValueError(
+            "radiative_rates_type type unknown - %s",
+            config.plasma.radiative_rates_type,
+        )
```
Comment (wkerzendorf) [inline tardis/plasma/standard_plasmas.py:152]:
```suggestion
        raise ValueError(f"radiative_rates_type type unknown - {config.plasma.radiative_rates_type}")
```

**Pull request merged**